compiler/packages/babel-plugin-react-compiler/src/ReactiveScopes/MemoizeExistingUseMemos.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { CompilerError } from "../CompilerError";
+import {
+  BasicBlock,
+  HIRFunction,
+  Identifier,
+  makeInstructionId,
+  ReactiveScope,
+} from "../HIR";
+import { eachTerminalSuccessor } from "../HIR/visitors";
+import {
+  collectMutableOperands,
+  mergeLocation,
+} from "./InferReactiveScopeVariables";
+
+export function memoizeExistingUseMemos(fn: HIRFunction): void {
+  visitBlock(fn, fn.body.blocks.get(fn.body.entry)!, null, new Map());
+}
+
+let ctr = 0;
+function nextId(): number {
+  return ctr++;
+}
```

Comment:
can we make this not a module-global, so that the count resets? seems like we should be able to just call env.nextScopeId as soon as we encounter a StartMemoize instruction, since we know that will map 1:1 with the scopes we create?